listenbrainz/webserver/static/js/src/PinnedRecordingCard.tsx

```diff
@@ -0,0 +1,240 @@
+  PinnedRecordingCardState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+
+  constructor(props: PinnedRecordingCardProps) {
+    super(props);
+    const currentlyPinned = this.determineIfCurrentlyPinned();
+    this.state = {
+      currentlyPinned,
+      isDeleted: false,
+    };
+  }
+
+  determineIfCurrentlyPinned = (): Boolean => {
+    const { PinnedRecording } = this.props;
+    const pinnedUntilTime: Date = new Date(PinnedRecording.pinned_until * 1000);
+
+    // invalid date
+    if (Number.isNaN(pinnedUntilTime.getTime())) {
+      return false;
+    }
+    const currentTime: Date = new Date();
+
+    if (currentTime < pinnedUntilTime) {
+      return true;
+    }
+
+    return false;
+  };
+
+  handleError = (error: string | Error, title?: string): void => {
+    const { newAlert } = this.props;
+    if (!error) {
+      return;
+    }
+    newAlert(
+      "danger",
+      title || "Error",
+      typeof error === "object" ? error.message : error
+    );
+  };
+
+  unpinRecording = async () => {
+    const { PinnedRecording, isCurrentUser, newAlert } = this.props;
+    const { APIService, currentUser } = this.context;
+
+    if (isCurrentUser && currentUser?.auth_token) {
+      try {
+        const status = await APIService.unpinRecording(currentUser.auth_token);
+        if (status === 200) {
+          this.setState({ currentlyPinned: false });
+          newAlert(
+            "success",
+            `You unpinned a recording.`,
+            `${PinnedRecording.track_metadata.artist_name} - ${PinnedRecording.track_metadata.track_name}`
+          );
+        }
+      } catch (error) {
+        this.handleError(error, "Error while unpinning recording");
+      }
+    }
+  };
+
+  deletePin = async (pinID: number) => {
+    const { isCurrentUser } = this.props;
+    const { APIService, currentUser } = this.context;
+
+    if (isCurrentUser && currentUser?.auth_token) {
+      try {
+        const status = await APIService.deletePin(
+          currentUser.auth_token,
+          pinID
+        );
+        if (status === 200) {
+          this.setState({ currentlyPinned: false });
+          this.setState({ isDeleted: true });
+        }
+      } catch (error) {
+        this.handleError(error, "Error while deleting pin");
+      }
+    }
+  };
+
+  render() {
+    const { PinnedRecording, isCurrentUser, userName } = this.props;
+    const { currentlyPinned, isDeleted } = this.state;
+    const { track_name } = PinnedRecording.track_metadata;
+    const { artist_name } = PinnedRecording.track_metadata;
+
+    return (
+      <Card
+        className={`pinned-recording-card row${
+          currentlyPinned ? " currently-pinned" : ""
+        }
+        ${isDeleted ? " deleted" : ""}`}
+      >
+        <div className={`${isCurrentUser ? " col-xs-9" : " col-xs-12"}`}>
+          <MediaQuery minWidth={768}>
+            <div className="col-xs-8">
```

Comment:
maybe we could create separate components for browser and phone? That would make this file quite a lot smaller and easier to read.